src/builder.rs

```diff
@@ -51,11 +52,47 @@ use crate::values::{
 
 use crate::{AtomicOrdering, AtomicRMWBinOp, FloatPredicate, IntPredicate};
 
+use std::cell::Cell;
 use std::marker::PhantomData;
 
+#[derive(Debug, PartialEq, Clone, Copy)]
+enum PositionState {
+    NotSet,
+    Set,
+}
+
+/// Errors that can be generated by the Builder. All `build_*` methods return a `Result<_, BuilderError>`, which must be handled.
+#[derive(Error, Debug, PartialEq, Eq)]
+pub enum BuilderError {
+    #[error("Builder position is not set")]
+    UnsetPosition,
+    #[error("GEP error: does not point to a struct or index is out of bounds")]
+    GEPError,
```

Comment:
Does this solve #213?

Reply:
I don't think so. The `GEPError` I added just replaces an `Err<'static str>` that was returned before #436 was merged. Could a `BuilderError` or something similar could be returned for #214?